sanicbot/core/cogs.py

```diff
@@ -0,0 +1,45 @@
+import aiohttp
+from discord.ext import commands
+from discord.ext.commands import Context
+
+from sanicbot.core.utils import failure_message, success_message, read_file
+
+
+class GitCog(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.org_url = f'https://github.com/sanic-org/'
+
+    async def lookup(self, number: int, repo):
+        """
+        Lookup the issue and check if it exists within a repo found in the Sanic organization. Redirection to
+        retrieve pull request is automatic due to the fact that Github issues them sequentially.
+
+        :param number: Issue number
+
+        :param repo: Repository of issue being looked up
+
+        :return: url, response_code
+        """
+        url = f'https://github.com/sanic-org/{repo}/issues/{number}'
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url) as response:
+                return str(response.url), response.status
+
+    @commands.command(aliases=['git', 'g'])
+    async def get_issue_or_pull(self, ctx: Context, number: int, repo: str = 'sanic'):
```

Comment:
This is a little confusing. A `get_` method I would expect to have a return value. It seems this does not return anything, and instead sends one of two message back to Discord. I would consider renaming this so it is more clear what it is doing.

Reply:
Is there a word I can use to define both pull requests and issues? Another method I was thinking is retrieve_git() or retrieve_issue_or_pull().

Reply:
> Note: GitHub's REST API v3 considers every pull request an issue, but not every issue is a pull request. For this reason, "Issues" endpoints may return both issues and pull requests in the response. You can identify pull requests by the pull_request key. Be aware that the id of a pull request returned from "Issues" endpoints will be an issue id. To find out the pull request id, use the "List pull requests" endpoint.

Just call them "issues"